src/devices/Ccs811/Ccs811Sensor.cs

```diff
@@ -0,0 +1,518 @@
+            temperatureCelsius += 25;
+            temperatureCelsius = Math.Max(temperatureCelsius, 0);
+            temperatureCelsius = Math.Min(temperatureCelsius, 127);
+            ConvertForEnvironement(temperatureCelsius, environment.Slice(2, 2));
+            WriteRegister(Register.ENV_DATA, environment);
+        }
+
+        private void ConvertForEnvironement(double toConvert, Span<byte> converted)
+        {
+            // Format is 7 bits for the integer part and 9 bits for the decimal one
+            byte integerPart = (byte)toConvert;
+            double decimalPart = toConvert - integerPart;
+            converted[0] = (byte)(integerPart << 1);
+            // There a 9 bits with fractions so we have to sample in 1/512 = 0.001953125
+            uint decimalPartUint = ((uint)(decimalPart / 0.001953125)) & 0x1FF;
+            converted[0] = (byte)(converted[0] | (decimalPartUint >> 8));
+            converted[1] = (byte)(decimalPartUint & 0xFF);
+        }
+
+        /// <summary>
+        /// Set the threshold for the equivalent CO2. The pinInterrupt should be existing so
+        /// interruptions are activated. If not, then the function will return false
+        /// </summary>
+        /// <param name="lowEquivalentCO2">The low value for the threshold in ppm</param>
+        /// <param name="highEquivalentCO2">The high value for the threshold in ppm</param>
+        /// <returns>True if success</returns>
+        /// <remarks>Difference between the low and high value should be more than 50. This is called
+        /// the hysteresis value.</remarks>
+        public bool SetThreshold(ushort lowEquivalentCO2, ushort highEquivalentCO2)
+        {
+            if (_pinInterruption < 0)
+            {
+                return false;
+            }
+
+            if (lowEquivalentCO2 > highEquivalentCO2)
+            {
+                var temp = highEquivalentCO2;
+                highEquivalentCO2 = lowEquivalentCO2;
+                lowEquivalentCO2 = temp;
+            }
+
+            if (highEquivalentCO2 - lowEquivalentCO2 < 50)
+            {
+                throw new ArgumentException($"{highEquivalentCO2}-{lowEquivalentCO2} should be more than 50");
+            }
+
+            Span<byte> toSend = stackalloc byte[4];
+            BinaryPrimitives.WriteUInt16BigEndian(toSend.Slice(0, 2), lowEquivalentCO2);
+            BinaryPrimitives.WriteUInt16BigEndian(toSend.Slice(2, 2), highEquivalentCO2);
+            WriteRegister(Register.THRESHOLDS, toSend);
+            // Activate the interrupt threshold as well
+            byte mode = ReadRegister(Register.MEAS_MODE);
+            mode |= 0b0000_0100;
+            WriteRegister(Register.MEAS_MODE, mode);
+
+            return !Status.HasFlag(Status.ERROR);
+        }
+
+        /// <summary>
+        /// Dispose the sensor
+        /// </summary>
+        public void Dispose()
+        {
+            _running = false;
```

Comment:
you might also want to wait until worker is done or double sure that it will not throw if controller is disposed